cmd/tink-worker/internal/registry.go

```diff
@@ -64,8 +74,17 @@ func (r *RegistryConnDetails) pullImage(ctx context.Context, cli *client.Client,
 		return errors.Wrap(err, "DOCKER PULL")
 	}
 	defer out.Close()
-	if _, err := io.Copy(os.Stdout, out); err != nil {
```

Comment:
Hm, I think I don't understand. so `cli.ImagePull` does not return error when you run out of disk space? This seems like a bug to me.

Reply:
yes, it is a bug. I think it is probably with docker. 

As of now, this change gives us leverage to bypass the error gracefully. But eventually, a proper fix on the go-client or docker would be required.

Reply:
IMO, i would have personally preferred the `io.Copy` catching that error. 

On our last syncup call me and @thebsdbox discussed this in details, so we thought to have a workaround ready at least for now. 

I also, think an alternative way could be to document this and ask the user to increase the disk space when this error is encountered.

Reply:
Has this been reported upstream? I'm also the user of `cli.ImagePull`, so I'm curious under what conditions it may fail, but not return error.

Reply:
I think probably the issue is with `io.Copy` writing stream to stdout instead of `cli.ImagePull`.

Reply:
`io.Copy` happens after `cli.ImagePull` error checking. If Image pulling fail, I'd expect no logs will be printed.

Reply:
The image is pulled in streams and read by `io.Copy` in chunks. Once one of the chunk is read EOF is generated. When read of second chunk fails, the interrupt is either not generated or go compiler doesn't receives it, so in a way for go error never occurred and that is the reason why go is not sending the error back. I have pasted the logs in the issue #382 .

Reply:
I'm not sure what are you referring to as "go compiler", but I don't think the compiler is involved in here. Maybe you meant _Go runtime_? 

So also looking at https://github.com/moby/moby/issues/36253, it seems `cli.ImagePull` does not really return error when pulling image fails, but streams the possible errors via the reader. This is hell-confusing API :smile:

Reply:
I created https://github.com/moby/moby/issues/42126, maybe they improve it somehow...